### PR TITLE
Add responsive filters

### PR DESCRIPTION
**Pull request by Pacific89**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Clean commits
- [ ] No warnings during install
- [ ] Added to release notes

### CODE REVIEW

File: src/components/ApplicationHeader.vue [hunk 1]
```diff
@@ -220,6 +253,26 @@ export default {
...
         return filtersActive.length
       }
+    },
+    loadOptions (filter) {
+      /**
+       * Each time a filter is expanded (shown=True) this function is triggered
+       * GetUpdateFilter checks how many search results each of the filter options will generate
+       */
+      const filterName = filter.name
+      // if (this.lastUpdatedFilter === filterName) {
```
Comment (jelmerveen) [inline src/components/ApplicationHeader.vue:263]: please remove unused code :)

File: src/components/ApplicationHeader.vue [hunk 2]
```diff
@@ -220,6 +253,26 @@ export default {
...
+      //   console.log('Not Updating')
+      //   return 0
+      // }
+      const activeFilters = this.activeFilters
+      const activation = true
```
Comment (jelmerveen) [inline src/components/ApplicationHeader.vue:268]: you could inline these constants like { activation: true }

File: src/store/actions.js [hunk 3]
```diff
@@ -55,6 +55,90 @@ export default {
...
+     * Construct query for all filter options of the currently expanded filter
+     * query the results using molgenis API and use "total" attribute to create a list
+     * of filter options: If result.total > 0 : append the corresponding filter option to the reduced list
+     * if result.total == 0, exclude the specific option (dont display it for corresponding filter)
+     */
+    const filterCheckPromises = []
+    const reducedFilterOptions = []
+
+    if (Object.keys(activeFilters).length === 0) {
+      console.log('Resetting...')
```
Comment (jelmerveen) [inline src/store/actions.js:72]: remove all debugging console logs

File: src/store/actions.js [hunk 4]
```diff
@@ -55,6 +55,90 @@ export default {
...
+      commit('ResetFilterLoading')
+      commit('ResetFilterOptionsOverride', { filterName, reducedFilterOptions })
+      return 0
+    }
+    // exclude active filters from current expanded filter
+    const activeFilterNames = Object.keys(activeFilters).filter(e => e !== filterName)
+    let url = '/api/data/eu_bbmri_eric_collections?size=1&filter=id&q='
+    // iterate over all active filter options and create query from current selection(s)
```
Comment (jelmerveen) [inline src/store/actions.js:80]: comments please in /** */ format, // is for todos

File: src/store/actions.js [hunk 5]
```diff
@@ -55,6 +55,90 @@ export default {
...
+        for (const option of activeFilters[name]) {
+          orQueryString.push(name + '=in=(' + option + ')')
+        }
+        const orOptions = orQueryString.join(';')
+        url = url + orOptions + ';'
+      } else {
+        url = url + name + '=in=(' + activeFilters[name] + ');'
+      }
+    }
+    console.log('FilterLoadingDict:')
```
Comment (jelmerveen) [inline src/store/actions.js:95]: console.logs

File: src/store/actions.js [hunk 6]
```diff
@@ -55,6 +55,90 @@ export default {
...
+    console.log(state.filterLoadingDict)
+    // check constructed URL for changes (compared to last udate)
```
Comment (jelmerveen) [inline src/store/actions.js:97]: comment notation

File: src/store/actions.js [hunk 7]
```diff
@@ -55,6 +55,90 @@ export default {
...
+    if (url === state.filterLoadingDict[filterName]) {
+      console.log('Not Updating...')
+      return 0
+    } else {
+      commit('SetFilterLoading', { filterName })
+    }
+    const lastBaseQuery = url
+    // const activeOptions = activeFilters[name].join(operator === 'OR' ? ',' : ';')
```
Comment (jelmerveen) [inline src/store/actions.js:105]: comment ? or dead code? or both :)

File: src/store/actions.js [hunk 8]
```diff
@@ -55,6 +55,90 @@ export default {
...
+    // .join(operator === 'OR' ? ',' : ';')
+    // iterate over options of the ONE filter that is currently expanded and construct query for each option
+    for (const option of state.filterOptionDictionary[filterName]) {
+      const filterOption = option.value
+      const optionString = filterName + '=in=(' + filterOption + ')'
+      // append URLs to list, use Promise.all on list
```
Comment (jelmerveen) [inline src/store/actions.js:111]: comment notation

File: src/store/actions.js [hunk 9]
```diff
@@ -55,6 +55,90 @@ export default {
...
+      const response = api.get(url + optionString)
+      // if (response_.total > 0) {
```
Comment (jelmerveen) [inline src/store/actions.js:113]: dead code

File: src/store/actions.js [hunk 10]
```diff
@@ -55,6 +55,90 @@ export default {
...
+    }
+
+    if (activeFilters[filterName]) {
+      for (const activeOption of activeFilters[filterName]) {
+        if (!reducedFilterOptions.includes(activeOption)) {
+          reducedFilterOptions.push(activeOption)
+        }
+      }
+    }
+    // for (const keep of activeFilters[filterName]) {
```
Comment (jelmerveen) [inline src/store/actions.js:126]: dead code

File: src/store/getters.js [hunk 11]
```diff
@@ -132,6 +132,12 @@ export default {
    * Get map of active filters
    */
   activeFilters: state => state.filters.selections,
+  filterLoadingDict: state => state.filterLoadingDict,
```
Comment (jelmerveen) [inline src/store/getters.js:135]: please do not make these passthrough getters, just use the mapState[''] function in the computed section of the component

This just adds more code to test, which has no logic.

File: src/store/mutations.js [hunk 12]
```diff
@@ -86,6 +86,40 @@ export default {
...
+      Vue.set(state, 'filterExpanded', 'None')
+    }
+  },
+  ResetFilterLoading (state) {
+    const filterLoadingDict = {}
+    const filters = state.filterFacets.map(fd => fd.name)
+    for (const f of filters) {
+      filterLoadingDict[f] = 'None'
+    }
+    console.log('Resetting')
```
Comment (jelmerveen) [inline src/store/mutations.js:105]: console

File: src/store/mutations.js [hunk 13]
```diff
@@ -86,6 +86,40 @@ export default {
...
+  SetUpdateFilter (state, { filterName, reducedFilterOptions, lastBaseQuery }) {
+    // initialize empty list for reduced set of filter options
+    state.filterOptionsOverride[filterName] = []
+    // set reduced filter options for corresponding filter (filerName)
+    Vue.set(state, 'filterOptionsOverride', { ...state.filterOptionsOverride, ...{ [filterName]: reducedFilterOptions } })
+    Vue.set(state, 'filterLoading', 'None')
+    Vue.set(state, 'lastUpdatedFilter', filterName)
+    Vue.set(state, 'lastBaseQuery', lastBaseQuery)
+    Vue.set(state, 'filterLoadingDict', { ...state.filterLoadingDict, ...{ [filterName]: lastBaseQuery } })
+    // state.filterOptionsOverride[filterName] = reducedFilterOptions
```
Comment (jelmerveen) [inline src/store/mutations.js:121]: dead code

File: src/store/mutations.js [hunk 14]
```diff
@@ -96,12 +130,27 @@ export default {
...
+
+    allStandards.forEach((standard) => {
+      qualityStandardsDictionary[standard.label] = standard.description || ''
+    })
+
+    state.qualityStandardsDictionary = qualityStandardsDictionary
+  },
   SetFilterOptionDictionary (state, { filterName, filterOptions }) {
-    /** only cache it once */
+    // only cache it once
```
Comment (jelmerveen) [inline src/store/mutations.js:149]: comment

**Review by jelmerveen — CHANGES_REQUESTED** [review]

File: src/store/getters.js (hunk 11 above)
Comment (Pacific89) [reply src/store/getters.js:135]: removed the getters but kept "activeFilters" as it is (i i dont think that i wrote this getter...)

**Comment by jelmerveen** [comment]
close because the diff is in a limbo state

**Pull request closed without merging**